### PR TITLE
Error handling for deployment scripts `deploy` subcommand

### PR DESCRIPTION
**Pull request by abulenok**
Partially solves #1361

## Introduced changes

- `deploy` from `sncast_std` now returns `Result<DeployResult, ScriptCommandError>`
- Added tests for `deploy` error handling in deployment scripts

## Checklist

- [x] Linked relevant issue
- [ ] Updated relevant documentation
- [ ] Added relevant tests
- [ ] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/sncast/tests/data/scripts/deploy/src/invalid_class_hash.cairo [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    let deploy_result = deploy(
+        class_hash,
+        array![0x2, 0x2, 0x0],
+        Option::Some(salt),
+        true,
+        Option::Some(max_fee),
+        Option::Some(deploy_nonce)
+    )
+        .unwrap_err();
+
```
Comment (THenry14) [inline crates/sncast/tests/data/scripts/deploy/src/invalid_class_hash.cairo:22]: should we assert the error here as well?

Comment (THenry14) [reply crates/sncast/tests/data/scripts/deploy/src/invalid_class_hash.cairo:22]: probably applicable to other tests as well

File: crates/sncast/tests/data/scripts/deploy/src/invalid_calldata.cairo [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        class_hash,
+        array![0x2],
+        Option::Some(salt),
+        true,
+        Option::Some(max_fee),
+        Option::Some(deploy_nonce)
+    )
+        .unwrap_err();
+
+    println!("test");
```
Comment (THenry14) [inline crates/sncast/tests/data/scripts/deploy/src/invalid_calldata.cairo:25]:
```suggestion
```

File: crates/sncast/tests/data/scripts/deploy/src/invalid_class_hash.cairo [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    println!("test");
```
Comment (THenry14) [inline crates/sncast/tests/data/scripts/deploy/src/invalid_class_hash.cairo:23]:
```suggestion
```

**Review by THenry14 — APPROVED** [review]
nits

File: crates/sncast/tests/data/scripts/deploy/src/invalid_class_hash.cairo (hunk 1 above)
Comment (abulenok) [reply crates/sncast/tests/data/scripts/deploy/src/invalid_class_hash.cairo:22]: In some tests, I assert the error like here:
```
assert(
    ScriptCommandError::ProviderError(
        ProviderError::StarknetError(StarknetError::InvalidTransactionNonce)
    ) == deploy_result,
    'ohno'
)
```

But it becomes more complex when the enum variant contains a struct like here:
```
pub enum StarknetError {
  ContractError: ErrorData,
}

pub struct ErrorData {
    msg: ByteArray
}
...
```

Comment (abulenok) [reply crates/sncast/tests/data/scripts/deploy/src/invalid_class_hash.cairo:22]: but I guess it's enough to assert the output produced with `println!("{:?}", result)`

File: crates/sncast/tests/data/scripts/deploy/src/invalid_calldata.cairo (hunk 2 above)
Comment (abulenok) [reply crates/sncast/tests/data/scripts/deploy/src/invalid_calldata.cairo]: removed

File: crates/sncast/tests/data/scripts/deploy/src/invalid_class_hash.cairo (hunk 3 above)
Comment (abulenok) [reply crates/sncast/tests/data/scripts/deploy/src/invalid_class_hash.cairo]: removed

**Pull request merged**